kobo/worker/main.py

```diff
@@ -37,7 +37,10 @@ def main_loop(conf, foreground=False):
             log_level = logging.getLevelName(conf.get("LOG_LEVEL", "DEBUG").upper())
             kobo.log.add_rotating_file_logger(logger, log_file, log_level=log_level)
 
-        tm = TaskManager(conf=conf, logger=logger)
+        if not task_manager_class:
```

Comment:
Minor: in my opinion better written as:

```python
# at top of function
task_manager_class = task_manager_class or TaskManager
# ...then later to construct:
tm = task_manager_class(...)
```

You would then not run the risk that the args passed to TaskManager or task_manager_class could diverge over time as the code is refactored.  It would also avoid reducing the statement coverage, which I think your change is currently doing since no test passes a non-None task_manager_class.